README.md
Screen Calculator
=================

A simple calculator for various screen based calculations such as PPI, screen dimensions, pixel count, etc.


ChangeLog
1.4.8 ~ September 9th, 2013
	- Added a reset button to the presets menu
	- Minor style changes

1.4.7 ~ September 5th, 2013
	- Added fading transitions for preset items in preset menu
	- Added "Currently Selected Preset" caption
	- Minor icon changes
	- Minor style changes
	- Minor code cleanup

1.4.6 ~ August 16th, 2013
	- Fixed Chromebook Pixel diagonal measurment
	- Fixed the link to my site
	- Minor style changes
	- Some code cleanup

1.4.5 ~ June 21st, 2013
	- Updated for the new Koding

1.4.4.1 ~ June 15th, 2013
	- Removed update script and all association with it

1.4.4 ~ June 2nd, 2013
	- Added update notification script
	- Changed speed of presets manu fading out when you click on a preset

1.4.3 ~ May 30th, 2013
	- Fixed bug where the aspect ratio would default to 16:9
	- Fixed alignment of presets
	- Added tablets to presets menu
	- Renamed calculate.js file to script.js

1.4.2 ~ May 24th, 2013
	- Fixed a calculation error for total pixels
	- Made presets menu fade in and out slightly faster
	- Updated .manifest file

1.4.1 ~ May 21st, 2013
	- Removed source.html file in place of adding full calculate.js file
	- Removed script tags and jQuery code from index.coffee
	- Repositioned the presets menu
	- Redid the presets menu (now fades in rather than slides in)
	- New open and close icons

1.4 ~ May 18th, 2013
	- Added More Info section which includes:
		- Screen size in centimeters
		- Width and height of each inch on the screen in pixels
		- Square inches in the screen
		- Aspect ratio of the screen (may be buggy)
		- Total pixels in the screen
	- The calculate() function now runs on load
	- Added source.html file to show comments and script in unminifed state
	- Minor bug fix
	- Minor style changes
	- More comments

1.3 ~ May 11th, 2013
	- Fixed calculator styling in Firefox
	- Some minor style changes/additions
	- Added more comments
	- Allowed decimal points in the diagonal measurement input field
	- Added close icon to the presets menu
	- Added -prefix-free for styling

1.2.1 ~ May 5th, 2013
	- Bug fix: Running the app would cause Koding's CSS to be affected
	- Updated the GitHub repository URL

1.2 ~ April 28th, 2013
	- Slight redesign of the Presets menu
	- Added a few more presets
	- Made it so that only numbers are allowed in text boxes
	- Added horizontal and vertical arrows to screen size boxes
	- Added disclaimer that all measurements are taken in landscape
	- Moved credits to the left
	- Code clean up

1.1 ~ April 27th, 2013
	- Slight redesign
	- Sets your current resolution (not diagonal measurement) on load
	- Removed "Calculate" button in place of automatic calculation
	- Added presets menu

1.0 ~ April 26th, 2013
	- Initial Release

resources/calculate.js
$(document).ready(function() {

    // Sets the default values for the resolution
    $('#h_res').val(screen.width);
    $('#v_res').val(screen.height);

    // The calculator
    function calculate() {

        // Detects what you put in the fields
        var x = $('#h_res').val();
        var y = $('#v_res').val();
        var d = $('#diag').val();

        // Calculates the screen size
        var ratio = y / x;
        var devWidth = Math.sqrt( Math.pow(d,2) / ( 1 + Math.pow(ratio,2)) );
        var devHeight = devWidth * ratio;

        // Displays the result rounded to two decimal places
        $('#height').text(devHeight.toFixed(2) + '"');
        $('#width').text(devWidth.toFixed(2) + '"');

        // Calculates the PPI
        var area = devHeight * devWidth;
        var ppi = Math.sqrt(x * y / area);

        // Displays the result rounded to two decimal places
        $('#ppi').text(ppi.toFixed(2));
        
        // Calculates the width and height of each inch of the device screen in px
        var lengthX = Math.sqrt(ppi);
        var lengthY = Math.sqrt(ppi);
        
        // Displays the result
        $('#lengthX').text(lengthX.toFixed(2) + ' px');
        $('#lengthY').text(lengthY.toFixed(2) + ' px');
        
        // Calculates the screen size in centimeters
        var centiWidth = devWidth * 2.54;
        var centiHeight = devHeight * 2.54;
        
        // Displays the result
        $('#centiWidth').text(centiWidth.toFixed(2) + ' cm');
        $('#centiHeight').text(centiHeight.toFixed(2) + ' cm');
        
        // Displays square inches in the screen (calculated earlier)
        $('#area').html(area.toFixed(2) + ' square inches');
        
        // Calculates the amount of pixels there are in the screen
        var pixels = (x* y);
        
        // Displays the result
        $('#pixels').text(pixels + ' pixels');
        
        // Calculates the screens aspect ratio (kinda buggy)
        var aspectRatio = (x / y).toFixed(2);
        
        // Displays the result
		if (aspectRatio == 1.25) {
			$('#aspectRatio').text('5:4');
		} else if (aspectRatio == 1.33) {
			$('#aspectRatio').text('4:3');
		} else if (aspectRatio == 1.5) {
			$('#aspectRatio').text('3:2');
		} else if (aspectRatio == 1.6) {
			$('#aspectRatio').text('16:10');
		} else if (aspectRatio == 1.67) {
			$('#aspectRatio').text('5:3');
		} else if (aspectRatio == 1.71) {
			$('#aspectRatio').text('128:75');
		} else if (aspectRatio == 1.77 || 1.78) {
			$('#aspectRatio').text('16:9');
		} else if (aspectRatio == 2.33) {
			$('#aspectRatio').text('21:9');
		} else {
			$('#aspectRatio').text(aspectRatio);
        }
    }

    // Allows only numbers in the resolution input fields
    $('#h_res, #v_res').keypress(function (e) {
        if (e.which !=8 && e.which !==0 && (e.which < 48 || e.which > 57)) {return false}
    });

    // Allows only numbers and decimals in the diagonal measurement input field
    $('#diag').keypress(function (e) {
        if (e.which !=8 && e.which !==0 && (e.which < 46 || e.which > 57)) {return false}
    });

    // Runs the calculate function while typing
    $('input').on('keyup', function() {
        calculate();
    });

	// More Info section
	$('#more').click(function() {
		// Shows and hides the More Info section
		$('.more').stop().slideToggle();
		
		// Changes the text of the link
		if ($(this).text() == "Click to see more (useless) info...")
			$(this).text("Click to see less (useless) info...");
		else
			$(this).text("Click to see more (useless) info...");
	});
	
	$('.open').click(function() {
		$('.presets').fadeIn(200);
	});
	$('.close').click(function() {
		$('.presets').fadeOut(200);
	});

    // The preset menu
    $('li.preset').on('click', function() {

        // Adds the data attribute values of the presets to the calculator
        $('#h_res').val( $(this).data('h') );
        $('#v_res').val( $(this).data('v') );
        $('#diag').val( $(this).data('diag') );

        // Calls the calculate() function
        calculate();

        // Closes the menu
		$('.presets').fadeOut();
		
    });
    
    calculate();
});

resources/script.js
$(document).ready(function() {
    // Sets the default values for the resolution
    $('#h_res').val(screen.width);
    $('#v_res').val(screen.height);

    // The calculator
    function calculate() {
        // Detects what you put in the fields
        var x = $('#h_res').val();
        var y = $('#v_res').val();
        var d = $('#diag').val();

        // Calculates the screen size
        var ratio = y / x;
        var devWidth = Math.sqrt(Math.pow(d,2) / (1 + Math.pow(ratio,2)));
        var devHeight = devWidth * ratio;

        // Displays the result rounded to two decimal places
        $('#width').text(devWidth.toFixed(2) + '"');
        $('#height').text(devHeight.toFixed(2) + '"');

        // Calculates the PPI
        var area = devHeight * devWidth;
        var ppi = Math.sqrt(x * y / area);

        // Displays the result rounded to two decimal places
        $('#ppi').text(ppi.toFixed(2));

        // Calculates the width and height of each inch of the device screen in px
        var lengthX = Math.sqrt(ppi);
        var lengthY = Math.sqrt(ppi);

        // Displays the result
        $('#lengthX').text(lengthX.toFixed(2) + ' px');
        $('#lengthY').text(lengthY.toFixed(2) + ' px');

        // Calculates the screen size in centimeters
        var centiWidth = devWidth * 2.54;
        var centiHeight = devHeight * 2.54;

        // Displays the result
        $('#centiWidth').text(centiWidth.toFixed(2) + ' cm');
        $('#centiHeight').text(centiHeight.toFixed(2) + ' cm');

        // Displays square inches in the screen (calculated earlier)
        $('#area').html(area.toFixed(2) + ' square inches');

        // Calculates the amount of pixels there are in the screen
        var pixels = (x * y);

        // Displays the result
        $('#pixels').text(pixels + ' pixels');

        // Calculates the screens aspect ratio
        var aspectRatio = (x / y).toFixed(2);

        var result;

        // Displays the result
        if (aspectRatio == 1.25) {
            result = '5:4';
        } else if (aspectRatio == 1.33) {
            result = '4:3';
        } else if (aspectRatio == 1.5) {
            result = '3:2';
        } else if (aspectRatio == 1.51) {
            result = '3:2';
        } else if (aspectRatio == 1.6) {
            result = '16:10';
        } else if (aspectRatio == 1.67) {
            result = '5:3';
        } else if (aspectRatio == 1.71) {
            result = '128:75';
        } else if (aspectRatio == 1.77) {
            result = '16:9';
        } else if (aspectRatio == 1.78) {
            result = '16:9';
        } else if (aspectRatio == 2.33) {
            result = '21:9';
        } else {
            result = aspectRatio;
        }
        $('#aspectRatio').text(result);
    }

    // Allows only numbers in the resolution input fields
    $('#h_res, #v_res').keypress(function (e) {
        if (e.which !=8 && e.which !==0 && (e.which < 48 || e.which > 57)) {return false}
    });

    // Allows only numbers and decimals in the diagonal measurement input field
    $('#diag').keypress(function (e) {
        if (e.which !=8 && e.which !==0 && (e.which < 46 || e.which > 57)) {return false}
    });

    // Runs the calculate function while typing
    $('input').on('keyup', function() {
        calculate();
    });

    // More Info section
    $('#more').click(function() {
        // Shows and hides the More Info section
        $('.more').stop().slideToggle();

        // Changes the text of the link
        if ($(this).text() == 'Click to see more info...')
            $(this).text('Click to see less info...');
        else
            $(this).text('Click to see more info...');
    });

    // Opens and closes the presets menu
    $('.open').click(function() {
        $('.presets').fadeIn(200);
    });
    $('.close').click(function() {
        $('.presets').fadeOut(200);
    });

    // The preset menu
    $('li.preset').on('click', function() {
        // Adds the data attribute values of the presets to the calculator
        $('#h_res').val($(this).data('h'));
        $('#v_res').val($(this).data('v'));
        $('#diag').val($(this).data('d'));

        // Sets the text of the "Currently Selected Preset" span to that of the selected preset
        $('#selected_preset span').text($(this).text());

        // Calls the calculate function
        calculate();

        // Closes the menu
        $('.presets').fadeOut(200);
    });

    $('#reset').on('click', function() {
        // Sets the values for the resolution to the users resolution anbd resets the diagonal measurment
        $('#h_res').val(screen.width);
        $('#v_res').val(screen.height);
        $('#diag').val('');

        // Resets the "Currently Selected Present" span
        $('#selected_preset span').text('None');

        // Runs the calculate function
        calculate();

        // Closes the menu
        $('.presets').fadeOut(200);
    });

    // Runs the calculate function
    calculate();
});